src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -281,3 +281,23 @@ AdjustStateForFailure(TaskExecution *taskExecution)
 	taskExecution->dataFetchTaskIndex = -1; /* reset data fetch counter */
 	taskExecution->failureCount++;          /* record failure */
 }
+
+
+/*
+ * CheckIfSizeLimitIsExceeded checks if the limit is exceeded by intermediate
+ * results, if there is any.
+ */
+bool
+CheckIfSizeLimitIsExceeded()
+{
+	uint64 MaxIntermediateResultInBytes = 0;
+	if (!UseResultSizeLimit)
+	{
+		return false;
+	}
+
+	MaxIntermediateResultInBytes = MaxIntermediateResult * 1024L;
+
+	return (MaxIntermediateResult > -1 && TotalIntermediateResultSize >
+			MaxIntermediateResultInBytes);
```

Comment:
looking at this again, could also structure it like:

```
if (MaxIntermediateResult == -1)
{
    return false;
}

MaxIntermediateResultInBytes = MaxIntermediateResult * 1024L;
if (TotalIntermediateResultSize < MaxIntermediateResultInBytes)
{
    return false;
}

return true;
```

we tend to be a little verbose about if-statements to make it easier to set breakpoints in gdb or add log output.